### PR TITLE
Socket Initialization on user login

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

Socket connection needs to be initialized only after user login.

## How was this tested?

Automated tests

### CODE REVIEW

**Review by dottorblaster — APPROVED** [review]
LGTM

File: assets/js/lib/api/wanda.js [hunk 1]
```diff
@@ -13,4 +13,4 @@ export const getLastExecutionByGroupID = (groupID) =>
   wandaClient.get(`/api/checks/groups/${groupID}/executions/last`);
 
 export const getCatalog = (env) =>
-  wandaClient.get(`/api/checks/catalog`, { params: env })
+  wandaClient.get(`/api/checks/catalog`, { params: env });
```
Comment (fabriziosestito) [inline assets/js/lib/api/wanda.js:16]: shouldn't the linter catch this?

Comment (CDimonaco) [reply assets/js/lib/api/wanda.js:16]: It was a formatting issue, for some reason previous formatting tasks didn't catch 🤷‍♂️

Comment (fabriziosestito) [reply assets/js/lib/api/wanda.js:16]: @CDimonaco can we investigate? are we checking the formatting?

Comment (CDimonaco) [reply assets/js/lib/api/wanda.js:16]: Yes in the CI we have the correct step, I will investigate further.

File: assets/js/state/sagas/index.js [hunk 2]
```diff
@@ -154,8 +157,16 @@ function* initialDataFetch() {
...
-  yield takeLatest('user/setUserAsLogged', initialDataFetch);
+function* setupSocketEvents() {
+  const socket = initSocketConnection();
+  yield call(processChannelEvents, store, socket);
+}
+
+function* watchUserLoggedIn() {
+  yield all([
+    takeLatest('user/setUserAsLogged', initialDataFetch),
+    takeLatest('user/setUserAsLogged', setupSocketEvents),
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply assets/js/state/sagas/index.js:168]: The application will crash, it's an unrecoverable error (no sockets no frontend basically), for the initial data fetching the error handling is the same as all the api requests

**Pull request merged**